### ISSUE TITLE
`forEach` breaks on null values

### ISSUE THREAD

**Issue by runeh** [sdgluck/mewt#33]
Steps to reproduce:

```
> immutable = require('mewt')
[Function]
> a = immutable([{test: null}])
[ { test: null } ]
> a.forEach(e => {})
TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at k (/Users/runeh/Dropbox/prosjekter/mock-serenity/node_modules/mewt/lib/index.js:5:12)
    at clone (/Users/runeh/Dropbox/prosjekter/mock-serenity/node_modules/mewt/lib/index.js:29:14)
    at k.reduce (/Users/runeh/Dropbox/prosjekter/mock-serenity/node_modules/mewt/lib/index.js:34:13)
    at Array.reduce (native)
    at clone (/Users/runeh/Dropbox/prosjekter/mock-serenity/node_modules/mewt/lib/index.js:29:21)
    at k.reduce (/Users/runeh/Dropbox/prosjekter/mock-serenity/node_modules/mewt/lib/index.js:34:13)
    at Array.reduce (native)
    at clone (/Users/runeh/Dropbox/prosjekter/mock-serenity/node_modules/mewt/lib/index.js:29:21)
    at Proxy.args (/Users/runeh/Dropbox/prosjekter/mock-serenity/node_modules/mewt/lib/index.js:47:38)
```

I would expect `forEach` to not throw in this case. Also could be usefuk if `forEach` returns the immutable, so it can be chained.

**Comment by sdgluck** [comment]
I believe this is related to the `null` property value issue you reported a moment ago, because this works:

```
$ node
> immutable = require('./lib/index')
[Function]
> a = immutable([{ test: 'test' }])
[ { test: 'test' } ]
> a.forEach(v => console.log(v))
{ test: 'test' }
```

However I will leave this open until I'm sure.

As for `forEach` returning the array, that is probably not going to be possible within the (self-imposed) 1kb limit, but I would be interested to hear any solutions if you have them.

**Comment by runeh** [comment]
Ah, I wasn't really considering the size limit, I've been using this on the server side.

I can abuse `.map` if I really need chaining and forEaching, so should be fine without it. Might be useful to mention in the readme that `forEach` has not been augmented.

**Comment by sdgluck** [comment]
Fixed in #34. I'll take a look at updating the README too later. Thanks @runeh!
